File: src/com/dafttech/terra/game/world/items/ItemEntitySpawner.java
package com.dafttech.terra.game.world.items;

import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.entities.Player;
import com.dafttech.terra.game.world.entities.models.EntityLiving;

public abstract class ItemEntitySpawner extends Item {

    @Override
    public void update(float delta) {
    }

    @Override
    public boolean use(EntityLiving causer, Vector2 position) {
        return spawnEntity(causer, position);
    }

    public abstract boolean spawnEntity(EntityLiving causer, Vector2 position);
}

File: terra/src/main/scala/com/dafttech/terra/game/world/items/ItemEntitySpawner.scala
package com.dafttech.terra.game.world.items

import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.entities.models.EntityLiving

abstract class ItemEntitySpawner extends Item {
  override def update(delta: Float): Unit = ()

  override def use(causer: EntityLiving, position: Vector2d): Boolean = spawnEntity(causer, position)

  def spawnEntity(causer: EntityLiving, position: Vector2d): Boolean
}